src/components/Footer/Footer.tsx

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+import { FilterType } from '../../types/FilterType';
+import { deleteTodo } from '../../api/todos';
+import { TextError } from '../../types/TextError';
+
+type Props = {
+  typeOfFilter: string;
+  setTypeOfFilter: (filter: string) => void;
+  todos: Todo[];
+  setTodos: (todos: Todo[]) => void;
+  setError: (value: TextError) => void;
+  completedTodoList: Todo[];
+};
+
+export const Footer: React.FC<Props> = ({
+  typeOfFilter,
+  setTypeOfFilter,
+  todos,
+  setTodos,
+  setError,
+  completedTodoList,
+}) => {
+  const activeTodos = todos.filter(todo => !todo.completed);
+
+  const deleteFinishedTodos = async (finished: Todo[]) => {
+    try {
+      finished.forEach(async (todo) => {
+        await deleteTodo(todo.id);
+
+        setTodos([...activeTodos]);
```

Comment:
If the new state is computed using the previous state, you can pass a function to setTodos